src/store/index.js

```diff
@@ -1,35 +1,44 @@
 const state = () => ({
-  persons: [
-    { id: 1, name: 'My Name' }
-  ]
-})
-
+  persons: [],
+});
 
 const mutations = {
-  'ADD_PERSON' (state, payload) {
-    state.persons.push(payload)
+  ADD_PERSON(state, payload) {
+    state.persons.push(payload);
+  },
+  CHANGE_PERSON(state, payload) {
+    const newState = state.persons.map((person) =>
+      person.id === Number(payload.target.id)
+        ? { ...person, name: payload.target.value }
+        : person
```

Comment:
Aqui eu to vinculando o `person.id` lá no input com `v-bind=person`. Pra não ter que repassar o id